Source/JsonParser.swift

```diff
@@ -154,21 +154,33 @@ internal final class JsonDeserializer: Parser {
             return unescapeMapping[character] ?? character
         }
         
-        var length = 0 // 2...8
+        guard let surrogateValue = parseEscapedUnicodeSurrogate() else { return nil }
+
+        // two consecutive \u#### sequences represent 32 bit unicode characters
+        if nextChar == Char(ascii: "\\") && source[cur.advancedBy(2)] == Char(ascii: "u") {
+                advance(); advance()
+                guard let surrogatePairValue = parseEscapedUnicodeSurrogate() else { return nil }
+                
+                return UnicodeScalar(surrogateValue << 16 | surrogatePairValue)
+        }
+        
+        return UnicodeScalar(surrogateValue)
```

Comment:
Technically, this will still crash if you pass it a high surrogate without a low surrogate following. You probably want to check whether what you're receiving is really a high/low surrogate pair.